### PR TITLE
Maven publishing

### LINKED ISSUE #43: implement alternate publishing solution to jitpack

### PR DESCRIPTION
**Pull request by phoebe-lew**
closes #43

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    types: [ published ]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'corretto'
+          java-version: '11'
```
Comment (jiyoontbd) [inline .github/workflows/publish.yaml:19]: i thought we upgraded to java 18, i might be wrong tho 🤔

**Review by jiyoontbd — APPROVED** [review]

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (ALRubinger) [reply .github/workflows/publish.yaml:19]: Should be 17 LTS - 18 is not entering LTS. @jiyoontbd please let me know if this isn't the case across the board for Java-based runtimes at TBD

File: build.gradle.kts [hunk 2]
```diff
@@ -22,8 +24,7 @@ dependencies {
 }
 
 allprojects {
-  version = "0.6.0"
-  group = "tbdex"
+  group = "xyz.block"
```
Comment (ALRubinger) [inline build.gradle.kts:27]: LGTM for now, `groupId` for TBD may change

File: build.gradle.kts [hunk 3]
```diff
@@ -108,4 +147,58 @@ subprojects {
...
+          license {
+            name.set("The Apache License, Version 2.0")
+            url.set("https://github.com/TBD54566975/tbdex-kt/blob/main/LICENSE")
+          }
+        }
+        developers {
+          developer {
+            id.set("TBD54566975")
+            name.set("Block Inc.")
+            email.set("tbd-releases@tbd.email")
```
Comment (ALRubinger) [inline build.gradle.kts:177]: +1, good

**Review by ALRubinger — APPROVED** [review]
Naked eye review +1 to me, leaving to @andresuribe87 to validate working as expected.

What's the full GAV of this? `xyz.block:what:whatVersioningScheme`? Wasn't discerning the `artifactId` from this PR.

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (phoebe-lew) [reply .github/workflows/publish.yaml:19]: Depends on whether or not Cash is planning to use our SDKs, they have services still on 11 IIRC.

**Review by andresuribe87 — APPROVED** [review]

**Pull request merged**